app/controllers/api/v1/book_controller.rb

```diff
@@ -2,7 +2,20 @@ module Api
   module V1
     class BookController < ApiController
       def index
-        render_paginated Book.all
+        if params[:isbn].nil?
+          render_paginated Book.all
+        else
+          show_isbn
+        end
+      end
+
+      def show_isbn
+        response = OpenLibraryService.book_info(params[:isbn])
+        if response.nil?
+          render json: 'Error: bad ISBN code', status: :bad_request
+        else
+          render json: response
+        end
```

Comment:
You can transform this 5 lines in 2 with a `return render "zarasa" if "condition"`